Java/core/src/main/java/com/amazon/randomcutforest/RandomCutForestState.java

```diff
@@ -18,52 +18,55 @@
 import java.util.ArrayList;
 import java.util.List;
 
+import lombok.Data;
+
 import com.amazon.randomcutforest.executor.Sequential;
-import com.amazon.randomcutforest.sampler.CompactSamplerData;
+import com.amazon.randomcutforest.sampler.CompactSamplerState;
 import com.amazon.randomcutforest.sampler.Weighted;
 import com.amazon.randomcutforest.store.PointStoreDoubleData;
-import com.amazon.randomcutforest.tree.TreeData;
+import com.amazon.randomcutforest.tree.CompactRandomCutTreeState;
 
 /**
  * A class that encapsulates most of the data used in a RandomCutForest such
  * that the forest can be serialized and deserialized.
  */
-public class ForestState {
-
-    public String typeOfForest = null;
+@Data
```

Comment:
np: you can also use the @Builder it makes creating these objects easier to read

Reply:
I like using builders in place of constructors with long argument lists, but right now all the state classes have 0-argument constructors and are populated using setters. In this case, I don't think using builders would be a big readability gain, but I'm not opposed to adding them.